orca-keel/src/main/kotlin/com/netflix/spinnaker/orca/keel/task/ImportDeliveryConfigTask.kt

```diff
@@ -115,6 +116,8 @@ constructor(
           // ...but give users a hint about 401 errors from igor/scm
           if (e.response?.status == 401 && URL(e.url).host.contains("igor")) {
             UNAUTHORIZED_SCM_ACCESS_MESSAGE
+          } else if (e.response?.status == 400 && URL(e.url).host.contains("keel") && e.response!!.body.length() > 0) {
```

Comment:
it is safe to do `response!!` here? What do you loose by instead using `?` so we don't throw an NPE if the response does not exist? (lns 119 & 120)

Reply:
Yes because nullability is checked in the first condition (`e.response?.status == 400`). What I gain is that I don't have to add a nested `if` or similar to access `.length` on `e.response.body` (which might be null if I used the `?`).

Reply:
I'd rather another if or chain of `?` instead of `!!`. I don't think we can say for certain that when `e.response` is not null then `body` will not be null.

Reply:
It's the other way around, actually. The expression `e.response?.status == 400` will ensure `e.response` is not `null`. Given that `&&` is a short-circuit operator, the remaining expressions will never get evaluated if `e.response` is `null`. So, when we get to `e.response!!.body`, `e.response` is guaranteed not to be `null`.

I can change it, but wanted to make sure we're on the same page first.

Reply:
P.S. Actually, if you expand the block of code immediately above, you'll see that the only way we even get here is when the status code in the response is `400..499`, which means the response is never `null` here.